File: Backend/src/main/java/com/example/Backend/dto/ProductDTO.java
package com.example.Backend.dto;

import jakarta.validation.constraints.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.util.Map;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class ProductDTO {

    @NotBlank(message = "Product name is required")
    private String name;

    private String description;

    @NotNull(message = "Price is required")
    @DecimalMin(value = "0.0", inclusive = false, message = "Price must be greater than 0")
    private BigDecimal price;

    private Long categoryId;
    private String category; // Keep for backward compatibility or display

    private Long brandId;
    private String brand; // Keep for backward compatibility or display

    @NotNull(message = "Stock quantity is required")
    @Min(value = 0, message = "Stock cannot be negative")
    private Integer stockQuantity;

    private String imageUrl;

    private Map<String, Object> vehicleCompatibility;

    private Boolean featured = false;
}

File: mobile-app/CarPartsEcom/app/src/main/java/com/example/carpartsecom/data/remote/dto/ProductDto.kt
package com.example.carpartsecom.data.remote.dto

import com.example.carpartsecom.data.local.entities.ProductEntity

data class ProductResponse(
    val id: Long,
    val name: String,
    val price: Double,
    val category: String,
    val description: String,
    val stockQuantity: Int,
    val rating: Double,
    val imageUrl: String
)

// Mapper
fun ProductResponse.toEntity(): ProductEntity {
    return ProductEntity(
        id = this.id,
        name = this.name,
        price = this.price,
        category = this.category,
        description = this.description,
        stockQuantity = this.stockQuantity,
        rating = this.rating,
        imageUrl = this.imageUrl
    )
}
